packages/scaffold-config/src/frameworks.ts

```diff
@@ -131,7 +131,14 @@ export const WIZARD_FRAMEWORKS = [
     codeGenFramework: 'react',
     mountModule: 'cypress/react',
     supportStatus: 'alpha',
-    componentIndexHtml: componentIndexHtmlGenerator('<div id="__next_css__DO_NOT_USE__"></div>'),
+    /**
+     * Next.js uses style-loader to inject CSS and requires this element to exist in the HTML.
+     * @see: https://github.com/vercel/next.js/blob/5f3351dbb8de71bcdbc91d869c04bc862a25da5f/packages/next/build/webpack/config/blocks/css/loaders/client.ts#L24
+     */
+    componentIndexHtml: componentIndexHtmlGenerator([
+      `<!-- Used by Next.js to inject CSS. -->\n`,
+      `<div id="__next_css__DO_NOT_USE__"></div>`,
+    ].join(' '.repeat(8))),
```

Comment:
It would make more sense to have this logic in `componentIndexHtmlGenerator` since it is aware of the proper spacing required to produce a formatted html, but this is the only use case so not a big deal.

Reply:
Not sure I agree here - I think `componentIndexHtmlGenerator` should not be too smart, and we should be injecting the additional markup from the outside. I don't think it needs to know about the different frameworks we support.

I tried to do something like:

```js
componentIndexHtml: dedent`
  <!-- stuff -->
`
```

But it won't work, since you end up with a `dedent` string inside a `dedent` string, and it doesn't format as you'd expect. This was the only way I could use the injection API that was implemented by the original developer and maintain the spacing. Not ideal, but it is what it is 🤷